src/seraphis_core/jamtis_payment_proposal.cpp

```diff
@@ -258,24 +279,24 @@ void get_output_proposal_v1(const JamtisPaymentProposalSelfSendV1 &proposal,
     CHECK_AND_ASSERT_THROW_MES(proposal.type <= JamtisSelfSendType::MAX,
         "jamtis payment proposal self-send: unknown self-send type.");
 
-    // 2. enote ephemeral pubkey: xK_e = xr xK_3
+    // 2. enote ephemeral pubkey: D_e = xr D^j_base
     get_enote_ephemeral_pubkey(proposal, enote_ephemeral_pubkey_out);
 
-    // 3. derived key: xK_d = xr * xK_2
-    crypto::x25519_pubkey xK_d; auto xKd_wiper = auto_wiper(xK_d);
-    crypto::x25519_scmul_key(proposal.enote_ephemeral_privkey, proposal.destination.addr_K2, xK_d);
+    // 3. derived key: D^d_fa = xr * D^j_fa
+    crypto::x25519_pubkey dhe_fa; auto dhe1_wiper = auto_wiper(dhe_fa);
+    crypto::x25519_scmul_key(proposal.enote_ephemeral_privkey, proposal.destination.addr_Dfa, dhe_fa);
 
-    // 4. sender-receiver shared secret (selfsend): q = H_32[k_vb](xK_e, input_context)  //note: xK_e not xK_d
+    // 4. sender-receiver shared secret (selfsend): q = H_32[k_vb](D_e, input_context)  //note: D_e not D^d_vr
     rct::key q; auto q_wiper = auto_wiper(q);
     make_jamtis_sender_receiver_secret_selfsend(k_view_balance,
         enote_ephemeral_pubkey_out,
         input_context,
-        proposal.type,
+        is_jamtis_auxiliary_selfsend_type(proposal.type),
         q);
 
-    // 5. amount baked key (selfsend): H_32[k_vb](q)
+    // 5. amount baked key (selfsend): H_32(self_send_type, q)
```

Comment:
```suggestion
    // 5. amount baked key (selfsend): H_32[k_vb](q)
```
Domain separator stuff isn't expressed. Also, k_vb is used here.